src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,78 @@
+package duke;
+
+import duke.command.*;
+import duke.dukeException.NoTimeGivenException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Encapsulates the parsing logic of string commands
+ */
+class Parser {
+    /**
+     * returns the Command object that the user indicates through userInput
+     *
+     * @param userInput the String value that the user typed
+     * @param taskList  the current list of tasks that duke stores, encapsulated in TaskList class
+     * @param over      the boolean value that decides when to kill the program, encapsulated in custom wrapper class
+     * @return the Command object that can be executed
+     * @throws NoTimeGivenException if the user types the wrong format of time
+     */
+    public static Command parse(String userInput, TaskList taskList, Over over) throws NoTimeGivenException {
+        String[] words = userInput.split(" ");
+        String firstWord = words[0];
+        if (firstWord.equals("bye")) {
+            return new CommandBye(over);
+        } else if (firstWord.equals("list")) {
+            return new CommandList(taskList);
+        } else if (firstWord.equals("mark") && words.length == 2) {
+            int taskNo = Integer.parseInt(words[1]);
+            return new CommandMark(taskList, taskNo);
+        } else if (firstWord.equals("unmark") && words.length == 2) {
+            int taskNo = Integer.parseInt(words[1]);
+            return new CommandUnmark(taskList, taskNo);
+        } else if (firstWord.equals("todo")) {
+            String todoContent = words[1];
+            for (int i = 2; i < words.length; i++) {
+                todoContent = todoContent + " " + words[i];
+            }
+            return new CommandTodo(taskList, todoContent);
+        } else if (firstWord.equals("deadline")) {
+            String deadlineContent = words[1];
+            for (int i = 2; i < words.length; i++) {
+                if (words[i].equals("/by")) {
+                    break;
+                }
+                deadlineContent = deadlineContent + " " + words[i];
```

Comment:
Wouldn't it be better if you could group all of this logic into one method? It would make the code more succinct and easier to read. You could also do the same thing for the rest of the code.
```suggestion
            //String deadlineContent = words[1];
            //for (int i = 2; i < words.length; i++) {
            //    if (words[i].equals("/by")) {
            //        break;
            //   }
            //    deadlineContent = deadlineContent + " " + words[i];
                deadlineContent = makeDeadlineContent(words);
```

Reply:
Yes!